### PR TITLE
readonly rootfs support

### PR DESCRIPTION
**Pull request by bergwolf**
A new command line option `--read-only` is added to `hyperctl run` and `hyperctl create` to enable readonly rootfs support. A new field `readonly` is added to container spec to enable it as well.

Readonly rootfs is enforced in two ways:
1. when specified, hyperstart will mount the container rootfs readonly
2. when specified, qemu will hotplug the device readonly for block storage. For aufs and overlay, hyperd will mount the rootfs readonly so that they are shared to qemu readonly, even though the 9p share dir is writable.

Needs: https://github.com/hyperhq/hyperstart/pull/314 https://github.com/hyperhq/runv/pull/511
Fixes: https://github.com/hyperhq/hyperd/issues/564

### CODE REVIEW

**Comment by feiskyer** [comment]
@bergwolf Any updates on this?

**Comment by bergwolf** [comment]
The hyperstart part needs rebase.

**Comment by bergwolf** [comment]
CI failures are because of `readonly` field parsing failure in hyperstart. We should merge  hyperhq/hyperstart#314 first and re-trigger CI here.

**Comment by bergwolf** [comment]
retest this please @hykins

**Comment by laijs** [comment]
it seems that the ro-mount for btrfs driver is not included.

**Comment by bergwolf** [comment]
hykins was running too slow (>40 mins) and timed out. retest this please @hykins

File: daemon/storage.go [hunk 1]
```diff
@@ -446,6 +446,12 @@ func (s *BtrfsStorage) PrepareContainer(containerId, sharedDir string, readonly
 	if err := syscall.Mount(btrfsRootfs, mountPoint, "bind", syscall.MS_BIND, ""); err != nil {
 		return nil, fmt.Errorf("failed to mount %s to %s: %v", btrfsRootfs, mountPoint, err)
 	}
+	if readonly {
```
Comment (laijs) [inline daemon/storage.go]: so there may be a security problem among these two mount operations. The early existing container can possible write to the roofs of this one.

Comment (bergwolf) [inline daemon/storage.go]: They are different subvolumes. It needs to break its own chroot sandbox to access another container's rootfs, no?

**Comment by laijs** [comment]
LGTM

**Pull request merged**